### PR TITLE
home-assistant: make parse-requirements.py print the percentage of supported components

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
Find out that right now 64.99% of the components are supported.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/home-assistant/parse-requirements.py [hunk 1]
```diff
@@ -221,6 +221,11 @@ def main() -> None:
         console = Console()
         console.print(table)
 
+    supported_components = reduce(lambda n, c: n + (build_inputs[c][1] == []),
+                                  components.keys(), 0)
+    n_components = len(components)
+    print(f"{supported_components} / {n_components} components supported, i.e."
+          f" {supported_components / n_components:.2%}")
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/parse-requirements.py:228]:
```suggestion
    total_components = len(components)
    print(f"{supported_components} / {total_components} components supported, i.e."
          f" {supported_components / total_components:.2%}")
```

**Comment by dotlambda** [comment]
Should we just drop the priniting of the missing dependencies instead? It's pretty useless.

**Comment by mweinelt** [comment]
Yeah, most interesting is the resulting file and the diff between releases, I guess.

**Comment by dotlambda** [comment]
Now the output is
```
$ mypy --ignore-missing-imports ./pkgs/servers/home-assistant/parse-requirements.py
Success: no issues found in 1 source file
Generating component-packages.nix for version 2021.6.4
         Outdated dependencies
┏━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━┓
┃ Package       ┃ Current    ┃ Wanted ┃
┡━━━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━┩
│ numpy         │ 1.20.2     │ 1.20.3 │
│ pydroid-ipcam │ 2021-06-01 │ 0.8    │
│ pyotgw        │ 2021-03-25 │ 1.1b1  │
│ sqlalchemy    │ 1.3.23     │ 1.4.13 │
└───────────────┴────────────┴────────┘
646 / 994 components supported, i.e. 64.99%
```

**Comment by mweinelt** [comment]
I don't really have a preference, do what you think is best.

**Comment by dotlambda** [comment]
I changed it and think this is good to go.

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
Thank you!

**Pull request merged**